CONTRIBUTING.md

```diff
@@ -1,425 +1,365 @@
-
-    ```shell
-    git push origin --delete my-feature-branch
-    ```
-
-* Check out the develop branch:
-
-    ```shell
-    git checkout develop -f
-    ```
-
-* Delete the local branch:
-
-    ```shell
-    git branch -D my-feature-branch
-    ```
-
-* Update your develop with the latest upstream version:
-
-    ```shell
-    git pull --ff upstream develop
-    ```
-
-[issue tracker]: https://github.com/marklogic/marklogic-data-hub/issues
-[.editorconfig]: http://editorconfig.org/
+# Contributing to MarkLogic Data Hub
+
+MarkLogic Data Hub welcomes new contributors. This document will guide you
+through the process.
+
+ - [Issues and Bugs](#found-an-issue)
+ - [Feature Requests](#want-a-feature)
+ - [Developing Data Hub](#developing-data-hub)
+ - [Submission Guidelines](#submission-guidelines)
+
+
+## Found an Issue?
+If you find a bug in the source code or a mistake in the documentation, you can help us by submitting an issue to our [GitHub Issue Tracker][issue tracker]. Even better, you can submit a Pull Request with a fix for the issue you filed.
+
+
+## Want a Feature?
+You can request a new feature by submitting an issue to our [GitHub Issue Tracker][issue tracker].  If you
+would like to implement a new feature then first create a new issue and discuss it with one of our
+project maintainers.
+
+
+## Developing Data Hub 
+
+This section describes how to build and test new features and fixes in Data Hub. This includes changes to the following subprojects in this project:
+
+- the Data Hub library in ./marklogic-data-hub
+- the Data Hub Gradle plugin in ./ml-data-hub-plugin
+- the One-UI web application in ./one-ui
+- the QuickStart web application in ./web
+
+### Prerequisites
+
+You need:
+
+- MarkLogic Server (see [Version Compatibility](https://docs.marklogic.com/datahub/refs/version-compatibility.html) for the correct version)
+- Java JDK 9 or later
+- Gradle 5.x or later
+
+### Testing changes to the Data Hub library
+
+The source code for the Data Hub library is in the ./marklogic-data-hub subproject. This project contains hundreds of 
+tests, written in JUnit and also written using [marklogic-unit-test](https://github.com/marklogic-community/marklogic-unit-test). 
+While developing, you'll create/modify/run tests to verify the changes that you're making. 
+
+To run the tests, you first need to deploy a test instance of Data Hub to MarkLogic. The configuration for this instance
+is in the ./marklogic-data-hub/gradle.properties; you can override these values in gradle-local.properties. Verify
+that the values for mlSecurityUsername and mlSecurityPassword are correct, as that user will be used to deploy the test
+instance of Data Hub. Also, ensure that you do not have a Data Hub instance already deployed to the MarkLogic that you 
+will connect to. Then, run the following Gradle task from the root project:
+
+    ./gradlew -Pskipui= bootstrap
```

Comment:
should we add -Pskipui=true? missing "true"

Reply:
Nope, as long as the property is defined, it'll achieve the desired effect.